### PR TITLE
Tickets/dm 39202

### PR DESCRIPTION
**Pull request by wvreeven**

### CODE REVIEW

File: tests/test_one_client_server.py [hunk 1]
```diff
@@ -83,20 +83,14 @@ async def connect_callback(self, server: tcpip.OneClientServer) -> None:
...
         await super().connect_callback(server)
 
-    def sync_connect_callback(self, server: tcpip.OneClientServer) -> None:
-        """A synchronous version of connect_callback.
-
-        TODO DM-37477: drop this method argument when we remove support
-        for synchronous connect_callback functions.
-        """
-        print(f"sync_connect_callback: connected={server.connected}")
+    def sync_connect_callback(self) -> None:
```
Comment (couger01) [inline tests/test_one_client_server.py:86]: Shouldn't this be removed?

File: tests/test_one_client_server.py [hunk 2]
```diff
@@ -298,15 +275,11 @@ async def test_read_write(self) -> None:
                     else:
                         raise
 
-    # TODO DM-37477: modify this test to expect a TypeError (and construct
-    # the OneClientServer directly, instead of calling create_server)
-    # once we drop support for sync connect_callback
     async def test_sync_connect_callback(self) -> None:
```
Comment (couger01) [inline tests/test_one_client_server.py:278]: Should this be renamed to test_async*?

**Review by couger01 — APPROVED** [review]
The code looks good but I have some confusion about some of the test functions.

File: tests/test_one_client_server.py (hunk 1 above)
Comment (wvreeven) [reply tests/test_one_client_server.py:86]: No, it shouldn't be removed. The purpose of this synchronous callback function is to support the unit test that shows that synchronous callback functions are not supported anymore.

Comment (wvreeven) [reply tests/test_one_client_server.py:86]: I added that sentence to the doc string of the function.

File: tests/test_one_client_server.py (hunk 2 above)
Comment (wvreeven) [reply tests/test_one_client_server.py:278]: I added that sentence to the doc string of the test case.

**Pull request merged**